src/repository.py

```diff
@@ -342,15 +396,37 @@ def update_branch(
         Returns:
             Repository client with the updated branch
         """
+        push_args = ["-u"]
+        if force:
+            push_args.append("-f")
+        push_args.extend([ORIGIN_NAME, self.current_branch])
+
         try:
+            # Create the branch if it doesn't exist
+            if push:
```

Comment:
What's the points of pushing before and after commiting?

Reply:
To create the branch, the PyGithub API doesn't allow for creating branches from specific commits, with a force git push it is allowed. Without pushing the branch first, GitHub actually allows unsigned commits through. By first pushing to create the branch and then pushing the commit, we catch that case and sign the commits in that case too

Reply:
Thx for the explanation